storage/queue.go

```diff
@@ -285,33 +311,55 @@ func (bq *baseQueue) processLoop(clock *hlc.Clock, stopper *stop.Stopper) {
 }
 
 func (bq *baseQueue) processOne(clock *hlc.Clock) {
+	// Load the system config.
+	cfg, err := bq.gossip.GetSystemConfig()
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
 	start := time.Now()
 	bq.Lock()
 	repl := bq.pop()
 	bq.Unlock()
-	if repl != nil {
-		now := clock.Now()
+
+	if repl == nil {
+		return
+	}
+
+	now := clock.Now()
+
+	if !bq.impl.acceptsUnsplitRanges() &&
```

Comment:
Let's split this code out into a method as it's duplicated between here and `MaybeAdd`.